src/router.rs

```diff
@@ -190,18 +208,26 @@ impl Router {
             url = Url::from_generic_url(generic_url).unwrap();
         }
 
-        self.recognize(&req.method, &path).and(
-            Some(IronError::new(TrailingSlash,
+        self.recognize(&req.method, &path).ok().and(
+            Some(IronError::new(RouterError::TrailingSlash,
                                 (status::MovedPermanently, Redirect(url))))
         )
     }
 
     fn handle_method(&self, req: &mut Request, path: &str) -> Option<IronResult<Response>> {
```

Comment:
I don't see this commit, the diff here looks the same to me. Did you forget to push?